src/test/resources/s3-valid-inbound-cases.xml

```diff
@@ -66,6 +66,22 @@
 		<integration:poller fixed-rate="1000"/>
 	</int-aws:s3-inbound-channel-adapter>
 
+	<int-aws:s3-inbound-channel-adapter
+			id="validInboundWithProxy"
+			bucket="TestBucket"
+			propertiesFile="testawscredentials.properties"
+			temporary-suffix=".temp"
+			local-directory="${foo.dir:/}"
+			remote-directory="remote"
+			accept-sub-folders="true"
+			max-objects-per-batch="100"
+			file-name-regex="[A-Za-z0-9]+\\.txt"
+			channel="inbound"
+			auto-startup="false"
+            proxyHost="localhost"
+            proxyPort="80">
```

Comment:
Sorry, I can't accept this fix at all...
I understand that you'd like to make some contribution, but right now it looks like you are in hurry.
The `ClientConfiguration` has many other options and my recommendation on your SO question sounds like:

> Well, looks like this is one of those critical requests for the Spring Integration AWS improvement: we really need to move our S3 adapters to the s-c-aws foundation. Meanwhile I see only the way to extend AbstractAmazonS3Operations with your logic to inject ClientConfiguration for the AmazonS3Client. Feel free to raise a JIRA ticket on the matter. 

The `<int-aws:s3-inbound-channel-adapter>` has `s3-operations` attribute. That's why you just need implement your own `AmazonS3Operations` with the `ClientConfiguration` and just inject it to the existing `<int-aws:s3-inbound-channel-adapter>` without any changes to the Framework.

From the Framework perspective we just need to add support for the `AmazonS3Client` which can be supplied with the `ClientConfiguration`. That was my request for JIRA.

Only these two options (`proxyHost` and `proxyPort`) really looks weird, just because we are able to configure more even without the changes to the framework...